dataclass_wizard/utils/typing_compat.py

```diff
@@ -376,7 +376,7 @@ def eval_forward_ref(base_type: FREF,
     return typing._eval_type(base_type, base_globals, _TYPING_LOCALS)
 
 
-def eval_forward_ref_if_needed(base_type: FREF,
+def eval_forward_ref_if_needed(base_type: typing.Union[typing.Type, FREF],
```

Comment:
As I understand here, the `if_needed` stands for the fact that the argument doesn't have to be FREF, we check it in the if below.

Reply:
totally right. I was also considering pulling out the `if` check and that way we could avoid an unnecessary function call most of the time, but I also do like it because all of that logic is abstracted away into a helper function. So i'm still debating what's the ideal approach, but you are right insofar as that `base_type` can either be a concrete type or a forward ref type, such as `str` for instance.